apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/SizedBufferingConsumer.cs

```diff
@@ -69,61 +64,25 @@ public static SizedBufferingConsumer<T> Create(IConsumer<T> consumer, Func<T, in
         public override void Dispose() => Flush();
 
         /// <inheritdoc />
-        protected override void ReceiveWithSemaphoreHeld(IEnumerable<T> items)
+        protected override bool ReceiveWithSemaphoreHeld(IEnumerable<T> items)
         {
             GaxPreconditions.CheckNotNull(items, nameof(items));
             foreach (T item in items)
             {
                 _size += _sizer(item);
                 _items.Add(item);
-                if (_size >= _bufferSize)
-                {
-                    FlushWithSemaphoreHeld();
-                }
             }
-        }
 
-        /// <inheritdoc />
-        protected override async Task ReceiveAsyncWithSemaphoreHeldAsync(
-            IEnumerable<T> items, CancellationToken cancellationToken = default)
-        {
-            GaxPreconditions.CheckNotNull(items, nameof(items));
-            foreach (T item in items)
-            {
-                _size += _sizer(item);
-                _items.Add(item);
-                if (_size >= _bufferSize)
-                {
-                    await FlushAsyncWithSemaphoreHeldAsync(cancellationToken).ConfigureAwait(false);
-                }
-            }
+            return _size >= _bufferSize;
```

Comment:
One change here is that if we get a *lot* of items here, it would previously have caused multiple flushes - now it'll only cause one. Is that okay?

Reply:
Yes, that's OK.
By default we only send one trace at a time to the `SizedBufferingConsumer`, I don't even think client code can configure the tracer so that this is not so. The method receives an `IEnumerable` because we implement the `IConsumer` interface, and some of its other implementors (i.e the consumers that this one flushes to) do receive several traces at the same time.